### DISCUSSION TITLE
Display FQDN hostname in prompt

### DISCUSSION THREAD

**Discussion by mlesin** [starship/starship#4707] [Q&A]
Can't find any way to show fully-qualified hostname in prompt
`hostname` command returns only host name on my host, while `hostname -f` returns what i really need: `host.example.com`
But I can't find how to explain this to starship.

Using starship on Ubuntu with fish shell

**Comment by theseal** [comment]
Think this is an Debian issue:
https://www.debian.org/doc/manuals/debian-reference/ch03.en.html#_the_hostname

I would also (sometimes) prefer to have the `fqdn` in the prompt instead of just the `hostname`.

Tried to read the source code and it seems like the library use to get the hostname have no way to include the domain part to the hostname (since libc probably returns it without domain). But that might be doable for someone how can write rust 😀

**Reply by davidkna** [reply]
The [default module config](https://starship.rs/config/#hostname) also strips everything after the first `.`, unless the `trim_at` option is changed.

**Reply by theseal** [reply]
Exactly, a good reason to at least support "both".

**Comment by ilium007** [comment]
2 years on... can we display fqdn yet?

**Comment by hleinders** [comment]
I'm using a workaround. The hostname module seems to depend on the output of $(hostname). I always have lots of shells open on my desktop and I wanted a more visible hint than the ssh symbol, so my local shell has a green background for user@host and a remote one is yellow. So I'm using two custom modules to replace both, $user and $hostname.

1. in my prompt:
format = """
$os\
${custom.userhost_local}\
${custom.userhost_remote}\
$directory\
....


with this custom modules:

[custom.userhost_local]
disabled = false
when = 'test -z "$SSH_TTY"'
command = 'echo "$USER@$HOSTNAME"'
style = "bg:color_local fg:color_black"
format = '[[>](bg:color_local fg:prev_bg) $output ]($style)'

[custom.userhost_remote]
disabled = false
when = 'test -n "$SSH_TTY"'
command = 'echo "$USER@$HOSTNAME"'
style = "bg:color_remote fg:color_black"
format = '[[>](bg:color_remote fg:prev_bg) $output ]($style)'

(the ">" is char U+E0B0, dosen't display here)

Some care has to be taken because $HOSTNAME is not alway set on a remote shell, so I've put something like 
"[ -z "$HOSTNAME" ] && set -gx HOSTNAME (hostname -s)" 
in my shell profile (should be "hostname -f" in your case).

So I get the local and remote prompts (my username is "hl", local host is "Edward", remote host is "Woody" :-) )


![Bildschirmfoto_20240919_095834](https://github.com/user-attachments/assets/8729d602-0c7f-49ae-bc65-274cbb99b17a)
![Bildschirmfoto_20240919_095847](https://github.com/user-attachments/assets/696faefe-f73d-4b1c-a32a-86a258525524)
